src/ds5/advanced_mode/json_loader.hpp

```diff
@@ -442,42 +443,56 @@ namespace librealsense
         return map;
     }
 
-    inline std::vector<uint8_t> generate_json(const preset& in_preset)
+    inline std::vector<uint8_t> generate_json(const device_interface& dev, const preset& in_preset)
     {
         preset_param_group p = in_preset;
         auto fields = initialize_field_parsers(p);
-
-        json j;
+        serialized_utilities::json_preset_writer preset_writer;
+        
+        preset_writer.set_device_info(dev);
+        
         for (auto&& f : fields)
         {
             if (f.second->is_duplicated) // Skip duplicated fields
                 continue;
 
             auto str = f.second->save();
             if (!str.empty()) // Ignored fields return empty string
-                j[f.first.c_str()] = str;
+                preset_writer.write_param(f.first.c_str(), str);
```

Comment:
It's not `set`/`get`? It's `write_param`/`find`

Reply:
Also, `write_param` will only work for `string` params, right? How do we make it so we can pass in any json content?

Reply:
1. IMO find is better than get because it should be used coupled with the end() function.
i.e: if (reader.find(key) != reader.end())

2. write param is a template the key is always a string but the value is templated.
`template < typename T >
void write_param(const std::string& key, T value)`

we can write int, float, string... which is all parameters under "parameters" section

Reply:
Agree about `find`.
Didn't see `write_param` was a template... but still: how do we use write_param with a json content (to make it hierarchical)? Will it work?

Reply:
It will take some effort that I am not sure is needed..
like adding a string parameters and parse it "device->first->second" for
```
device:{
first:{
second{
key:value
}}}
```